diff --git a/inc/inventorycomputerlib.class.php b/inc/inventorycomputerlib.class.php
index a05c7a6e8..c0b7511fe 100644
--- a/inc/inventorycomputerlib.class.php
+++ b/inc/inventorycomputerlib.class.php
@@ -1805,7 +1805,7 @@ class PluginFusioninventoryInventoryComputerLib extends PluginFusioninventoryInv
       $deviceBios       = new DeviceFirmware();
 
       $fwTypes = new DeviceFirmwareType();
-      $fwTypes->getFromDBByQuery("WHERE `name` = 'BIOS'");
+      $fwTypes->getFromDBByCrit(['name' => 'BIOS']);
       $type_id = $fwTypes->getID();
       $data['devicefirmwaretypes_id'] = $type_id;
 
