diff --git a/inc/commondbtm.class.php b/inc/commondbtm.class.php
index 7f8bc4fdb..788af9aa9 100644
--- a/inc/commondbtm.class.php
+++ b/inc/commondbtm.class.php
@@ -3388,7 +3388,7 @@ class CommonDBTM extends CommonGLPI {
    public final function searchOptions() {
       $options = [];

-      if (method_exists(get_class(), 'getSearchOptions')) {
+      if (method_exists(get_class($this), 'getSearchOptions')) {
          if (defined('TU_USER')) {
             throw new \RuntimeException('getSearchOptions must not be used!');
          }
