diff --git a/setup.php b/setup.php
index f5b24a143..413a52768 100644
--- a/setup.php
+++ b/setup.php
@@ -486,7 +486,7 @@ function plugin_version_fusioninventory() {
            'requirements'   => [
               'glpi' => [
                  'min' => '9.3',
-                  'max' => '9.4',
+                  'max' => '9.5',
                   'dev' => PLUGIN_FUSIONINVENTORY_OFFICIAL_RELEASE == 0
                ],
                'php' => [
@@ -523,11 +523,11 @@ function plugin_fusioninventory_check_prerequisites() {
 
    if (version_compare(GLPI_VERSION, '9.2-dev', '!=')
       && version_compare(GLPI_VERSION, '9.2', 'lt')
-      || version_compare(GLPI_VERSION, '9.3', 'ge')) {
+      || version_compare(GLPI_VERSION, '9.5', 'ge')) {
       if (method_exists('Plugin', 'messageIncompatible')) {
-         echo Plugin::messageIncompatible('core', '9.2', '9.3');
+         echo Plugin::messageIncompatible('core', '9.2', '9.5');
       } else {
-         echo __('Your GLPI version not compatible, require >= 9.2 and < 9.3', 'fusioninventory');
+         echo __('Your GLPI version not compatible, require >= 9.2 and < 9.5', 'fusioninventory');
       }
       return false;
    }
