diff --git a/inc/inventorycomputerlib.class.php b/inc/inventorycomputerlib.class.php
index c1f9c5a2f..0e55757b5 100644
--- a/inc/inventorycomputerlib.class.php
+++ b/inc/inventorycomputerlib.class.php
@@ -176,10 +176,10 @@ class PluginFusioninventoryInventoryComputerLib extends PluginFusioninventoryInv
                if ($ios->fields[$key] != $value) {
                   $same = false;
                   break;
+               } else {
+                  $input_os['_no_history'] = $no_history;
+                  $ios->add($input_os);
                }
-            } else {
-               $input_os['_no_history'] = $no_history;
-               $ios->add($input_os);
             }
             if ($same === false) {
                $ios->update(['id' => $ios->getID()] + $input_os);
