diff --git a/setup.php b/setup.php
index eed6660..1f065eb 100644
--- a/setup.php
+++ b/setup.php
@@ -209,7 +209,7 @@ function plugin_version_flyvemdm() {
       'requirements'   => [
          'glpi' => [
             'min' => $glpiVersion,
-            'max' => PLUGIN_FLYVEMDM_GLPI_MAX_VERSION,
+            'max' => '9.5',
             'dev' => PLUGIN_FLYVEMDM_IS_OFFICIAL_RELEASE == false,
             'plugins'   => [
                'fusioninventory',
