diff --git a/setup.php b/setup.php
index eed6660..79c56de 100644
--- a/setup.php
+++ b/setup.php
@@ -35,7 +35,7 @@ define('PLUGIN_FLYVEMDM_IS_OFFICIAL_RELEASE', false);
 // Minimal GLPI version, inclusive
 define('PLUGIN_FLYVEMDM_GLPI_MIN_VERSION', '9.2');
 // Maximum GLPI version, exclusive
-define('PLUGIN_FLYVEMDM_GLPI_MAX_VERSION', '9.4');
+define('PLUGIN_FLYVEMDM_GLPI_MAX_VERSION', '9.5');

 define('PLUGIN_FLYVEMDM_ROOT', GLPI_ROOT . '/plugins/flyvemdm');
